### PR TITLE
fix: define window.name

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes https://github.com/denoland/deno/issues/20750

This matches what browsers do: https://developer.mozilla.org/en-US/docs/Web/API/Window/name

In the future we might want to change the behavior to actually update
the process name, but that needs a bit of discussion regarding if
it needs a permission flag (that would make polyfiling `process.title`
setter really easy too).

### CODE REVIEW

File: cli/tests/unit/globals_test.ts [hunk 1]
```diff
@@ -128,3 +128,16 @@ Deno.test(function webApiGlobalThis() {
   assert(globalThis.CountQueuingStrategy !== null);
   assert(globalThis.ByteLengthQueuingStrategy !== null);
 });
+
+Deno.test(function windowNameIsDefined() {
+  assert(typeof globalThis.name === "string");
+  assert(name === "");
+  assert(window.name === name);
+  name = "foobar";
+  // @ts-ignore somehow TS doesn't understand that name === window.name
```
Comment (dsherret) [inline cli/tests/unit/globals_test.ts:137]: Use assertEquals instead? I feel like most of these should be using that anyway.

**Review by dsherret — APPROVED** [review]
LGTM

**Review by lucacasonato — COMMENTED** [review]
Mh I actually really dislike this global because folks often call variables `name`, and now if they forget to declare them with let / const first, setting/ reading will still work without throwing.

**Comment by bartlomieju** [comment]
@lucacasonato so what would you suggest to do here? I think `name` is the perfect place to implement process name setter as it already follows what browsers do.

**Comment by dsherret** [comment]
@bartlomieju shouldn't we deprecate `window`?

**Comment by bartlomieju** [comment]
I'm still not a fan of deprecating `window`, but if we're gonna go that route then it probably makes sense. Should I revert this PR?

**Comment by dsherret** [comment]
My guess would be that the source of the original issue is that the window global exists in the first place.

**Pull request merged**